Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -294,15 +336,27 @@ protected void setParent(int child, int parent) {
         }
     }
 
+    /**
+     * deletes a node from the stores and resets the connectivity information so
+     * that state is preserved
+     * 
+     * @param node the node to be deleted
+     */
     @Override
     protected void delete(Integer node) {
         delete(intValue(node));
     }
 
     protected void delete(int node) {
         if (isLeaf(node)) {
-            leafStore.delete(getLeafIndexForTreeIndex(node));
+            int index = getLeafIndexForTreeIndex(node);
+            leafStore.setPointIndex(index, PointStore.INFEASIBLE_POINTSTORE_INDEX);
```

Comment:
In code, no - but helped me in keeping the different indices a bit separate. The plan is to set different negative values to these "NULL"s so that it's easier to debug (should we ever need to)